ports/stm32/mboot/main.c

```diff
@@ -895,6 +895,9 @@ static int dfu_handle_tx(int cmd, int arg, int len, uint8_t *buf, int max_len) {
         buf[4] = dfu_context.state; // bState
         buf[5] = dfu_context.error; // iString
         return 6;
+    } else if (cmd == DFU_GETSTATE && len == 1) {
+        buf[0] = dfu_context.state; // bState
+        return 1;
```

Comment:
I thought for a minute there that this request should also clear the state, however from page 23 of: https://www.usb.org/sites/default/files/DFU_1.1.pdf

> 6.1.5 DFU_GETSTATE Request
> This request solicits a report about the state of the device. The state reported is the current state of the
> device with no change in state upon transmission of the response. The values specified in the bState
> field are identical to those reported in DFU_GETSTATUS

Reply:
Thanks for checking!